### PR TITLE
Update Pipeline Scripts (plus everything that affected)

### PR DESCRIPTION
**Pull request by DerekRoberts**
Bringing over existing WPS shared scripts.  Some changes to BC and DC for imagestreams, which are now separate for DEV and PROD.

### CODE REVIEW

**Comment by DerekRoberts** [comment]
Resolves #208
Resolves #216

**Comment by DerekRoberts** [comment]
Resolves #216

**Comment by DerekRoberts** [comment]
> Can you cherry pick BC, DC, and Jenkinsfile file changes from #32

@Kyubinhan Does that have anything required to get this PR through?  This PR is an subtree update ticket, so there's no new code created.

**Comment by DerekRoberts** [comment]
> Can you cherry pick BC, DC, and Jenkinsfile file changes from #32

That's a lot to pull over and the PR would benefit from some cleanup.  How far is that from being merged into PROD?  Then it'll be far easier to pull over.

**Comment by Kyubinhan** [comment]
@DerekRoberts I realize the pr got a lot of commits, so can you just copy the last version of 3 files that I mentioned from the pr?

**Comment by DerekRoberts** [comment]
> Don't the build.bc and deploy.dc have to be updated same as in [bcgov/wps-web#91](https://github.com/bcgov/wps-web/pull/91) ?

@Sybrand @Kyubinhan Yup!  In progress, working with Nithin now.

**Comment by DerekRoberts** [comment]
> @DerekRoberts I realize the pr got a lot of commits, so can you just copy the last version of 3 files that I mentioned above from the pr?

@Kyubinhan Done!

**Comment by Kyubinhan** [comment]
@DerekRoberts if you are going to use GHA for api, can you disable stages in Jenkins

**Comment by DerekRoberts** [comment]
> @DerekRoberts if you are going to use GHA for api, can you disable stages in Jenkins

@Kyubinhan That's the plan!  I'm mostly working with scripts locally right now, so Jenkins can wait.

**Comment by DerekRoberts** [comment]
@Sybrand @Kyubinhan @Nkuruba Can you please check the Lint and Unit Test stages to see if it's failing due to pipeline error or not having been linted enough?

**Comment by Sybrand** [comment]
> @Sybrand @Kyubinhan @Nkuruba @andrea-williams Can you please check the Lint and Unit Test stage to see if it's failing due to pipeline error or not having been linted enough?

@DerekRoberts there are a few reason:
- I think it's using python2, instead of the correct version. (on ubuntu you have to say python3).
- I'm not seeing where you're installing the dependancies using Pipenv/requirements.txt

If you look at the output of the test - you'll also see it's not actually running a single test.

File: .github/workflows/pr-dev.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    strategy:
+      matrix:
+        python-version: [3.6]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Python ${{ matrix.node-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.node-version }}
+      - run: python -m unittest
```
Comment (Sybrand) [inline .github/workflows/pr-dev.yml:21]: Ubuntu uses python2 by default.
`python3 -m unittest`

File: .github/workflows/pr-dev.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - run: python -m pip install pylint
+      - run: python -m pylint --rcfile=.pylintrc *.py **/*.py
```
Comment (Sybrand) [inline .github/workflows/pr-dev.yml:23]: This is probably going to fail without dependancies having been installed. Please refer to Dockerfile. There are a few strategies for doing this:

- install global using pipenv
- create requirements.txt then install from there.
- run with pipenv

I recommend going with the requirements.txt strategy, as that is going to give you the cleanest install closest to production.

**Comment by DerekRoberts** [comment]
@Sybrand Please confirm I was supposed to remove the "##### <START TEMPORARY!!!!" section of the Dockerfile now that unit tests and linting are in the pipeline.

**Comment by Sybrand** [comment]
> @Sybrand Please confirm I was supposed to remove the "##### <START TEMPORARY!!!!" section of the Dockerfile now that unit tests and linting are in the pipeline.

@DerekRoberts yep - there's a good argument to be made that should never have gone in there anyway.

**Comment by DerekRoberts** [comment]
Cleaning up before re-review.  I definitely let this sprawl.

**Comment by DerekRoberts** [comment]
@Sybrand @Kyubinhan @Nkuruba @andrea-williams PR scaled back to just script updates and is (really) ready!  If the comments are too messy I can fire up a fresh PR.

**Pull request merged**